client/go/util/array.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+// Author: arnej
+
+// generic utilities
+package util
+
+type Array[E comparable] []E
+
+func NewArray[E comparable](initialCapacity int) Array[E] {
+	return make([]E, 0, initialCapacity)
+}
+
+func ArrayOf[E comparable](elems []E) Array[E] {
+	return Array[E](elems)
+}
+
+func (arrayP *Array[E]) Append(elem E) {
+	*arrayP = append(*arrayP, elem)
+}
+
+func (arrayP *Array[E]) AppendElements(elems ...E) {
+	arrayP.AppendAll(elems)
+}
+
+func (arrayP *Array[E]) AppendAll(elemsToAppend []E) {
```

Comment:
Do we need both? If you have a slice you can unpack it to args, e.g. `AppendElements(myslice...)`